### PR TITLE
[movevm] more validation for sponsored transaction

### PR DESCRIPTION
**Pull request by davidiw**
Test if the account resource exists before attempting to create the account resource. This solves potential devx issues around gas fees and also prevents potential errors when the account exists but sequence number 0 isn't expensive enough.

updated one of the tests to explicitly test for this condition.

### CODE REVIEW

File: aptos-move/aptos-vm/src/aptos_vm.rs
Comment (movekevin) [inline aptos-move/aptos-vm/src/aptos_vm.rs:529]: Remove before merging

**Review by movekevin — APPROVED** [review]

File: aptos-move/aptos-vm/src/aptos_vm.rs
Comment (georgemitenkov) [inline aptos-move/aptos-vm/src/aptos_vm.rs:2299]: Should we propagate the error when get_resource returns Err?

Comment (davidiw) [reply aptos-move/aptos-vm/src/aptos_vm.rs:2299]: I'm not sure it would ultimately have any impact other than an earlier flag that we have some bizarre invariant violation... your call.

Comment (davidiw) [reply aptos-move/aptos-vm/src/aptos_vm.rs:2299]: note, here we didn't return the error: https://github.com/aptos-labs/aptos-core/blob/da2db7e9c467884be5226a360fb1cf5499a13598/aptos-move/aptos-vm/src/gas.rs#L127

so I'd be hesitant to adopt this policy now.

File: aptos-move/e2e-move-tests/src/tests/fee_payer.rs
Comment (ziaptos) [inline aptos-move/e2e-move-tests/src/tests/fee_payer.rs:63]: Do we need `SPONSORED_AUTOMATIC_ACCOUNT_V1_CREATION` flag also on?

**Review by ziaptos — COMMENTED** [review]
Some parts of the comments are outdated (diff updated while I was reviewing)

File: aptos-move/e2e-move-tests/src/tests/fee_payer.rs
Comment (davidiw) [reply aptos-move/e2e-move-tests/src/tests/fee_payer.rs:63]: hmm... this is mostly for comments, because the default behavior is it is on...

Comment (ziaptos) [reply aptos-move/e2e-move-tests/src/tests/fee_payer.rs:63]: Will the test be OK when the default behavior changes? when the SPONSORED_AUTOMATIC_ACCOUNT_V1_CREATION is off?

Comment (davidiw) [reply aptos-move/e2e-move-tests/src/tests/fee_payer.rs:63]: this is set in vm-genesis, if we change default flags, massive amounts of tests across the code base will start to fail not just in e2e-move-tests. We can do better here.

File: aptos-move/aptos-vm/src/aptos_vm.rs
Comment (davidiw) [reply aptos-move/aptos-vm/src/aptos_vm.rs]: I spent more time thinking about this, and while returning none and error should be similar, if we can actually just exit early it requires less reasoning, hence I fixed this to return the error.

**Review by ziaptos — APPROVED** [review]
Traffic Control: Clear for landing

File: aptos-move/e2e-move-tests/src/tests/fee_payer.rs
Comment (davidiw) [reply aptos-move/e2e-move-tests/src/tests/fee_payer.rs]: eliminated the ambiguity...

File: aptos-move/aptos-vm/src/aptos_vm.rs
Comment (igor-aptos) [inline aptos-move/aptos-vm/src/aptos_vm.rs:2319]: Not at the laptop, but don't we have exists_resource or similar method in the resolver?

(You can proceed with landing and cherry picking to 1.9 as is, maybe worth cleaning on main)

**Review by igor-aptos — APPROVED** [review]

File: aptos-move/aptos-vm/src/aptos_vm.rs
Comment (davidiw) [reply aptos-move/aptos-vm/src/aptos_vm.rs:2319]: `resource_exists_in_group` but that's more about the type and not the data, from what I recall.

Comment (igor-aptos) [reply aptos-move/aptos-vm/src/aptos_vm.rs:2319]: I was talking about `resource_exists` method defined here - https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/aptos-vm-types/src/resolver.rs#L54

`resource_exists_in_group` is for whether tag exists in resource group.

Comment (davidiw) [reply aptos-move/aptos-vm/src/aptos_vm.rs:2319]: ahh, neat!

Comment (igor-aptos) [reply aptos-move/aptos-vm/src/aptos_vm.rs:2319]: cool part about it, is that if transaction BlockSTM executes concurrently modifies the value (with resource existing before and after) - it will not be treated as a conflict. Where get_resource would need to re-execute - because it doesn't value is being ignored

Comment (davidiw) [reply aptos-move/aptos-vm/src/aptos_vm.rs:2319]: right, right, the only conflict might be if it is created or deleted, I imagine?

Comment (igor-aptos) [reply aptos-move/aptos-vm/src/aptos_vm.rs:2319]: yup :)

**Pull request merged**
